xds/src/main/java/io/grpc/xds/XdsClientImpl.java

```diff
@@ -380,6 +385,30 @@ public String toString() {
     return logId.toString();
   }
 
+  @Override
+  public void startSubscriberTimersIfNeeded(ServerInfo serverInfo) {
+    if (isShutDown()) {
+      return;
+    }
+
+    syncContext.execute(new Runnable() {
+      @Override
+      public void run() {
+        if (isShutDown()) {
+          return;
+        }
+
+        for (Map<String, ResourceSubscriber<?>> subscriberMap : resourceSubscribers.values()) {
+          for (ResourceSubscriber<?> subscriber : subscriberMap.values()) {
+            if (subscriber.serverInfo.equals(serverInfo) && subscriber.respTimer == null) {
```

Comment:
I looked into using `serverInfo.equals` here: `equals` is autogenerated in `AutoValue_Bootstrapper_ServerInfo` and assumes `equals` is defined for ` ChannelCredentials` (one of the members) and that does not seem to be the case. I don't know if that is a more generic problem and needs to be fixed there or should be fixed as part of this PR?

Reply:
The equals method is defined on java.lang.Object as the addresses being the same, so it just means that rather than a logical equality it must be the same actual object.

Reply:
> as the addresses being the same

So in this case we are okay to say that for `ChannelCredentials` equality is possible only when the 2 objects are exactly the same object.

Reply:
Yes